### PR TITLE
Use anyFormat for all icons

### LINKED ISSUE #1728: Some icons don't use anyFormat
I forgot to update some icon request in the full flavor, e.g. https://github.com/openhab/openhab-android/blob/master/mobile/src/full/java/org/openhab/habdroid/core/FcmMessageListenerService.kt#L129

_Originally posted by @mueller-ma in https://github.com/openhab/openhab-android/pull/1667#issuecomment-567001808_

### PR DESCRIPTION
**Pull request by mueller-ma**
Fixes #1728

Signed-off-by: mueller-ma <mueller-ma@users.noreply.github.com>

### CODE REVIEW

File: mobile/src/full/java/org/openhab/habdroid/core/FcmMessageListenerService.kt [hunk 1]
```diff
@@ -123,10 +126,15 @@ class FcmMessageListenerService : FirebaseMessagingService() {
...
+            val encodedIcon = Uri.encode(icon)
+            val suffix = when (getPrefs().getIconFormat()) {
+                IconFormat.Png -> "png"
+                IconFormat.Svg -> "svg"
+            }
             if (connection != null) {
                 try {
                     iconBitmap = connection.httpClient
-                            .get("images/$icon.png", timeoutMillis = 1000)
+                            .get("icon/$encodedIcon?&format=$suffix&anyFormat=true", timeoutMillis = 1000)
```
Comment (maniac103) [inline mobile/src/full/java/org/openhab/habdroid/core/FcmMessageListenerService.kt]: `&format` -> `format` (`&` isn't needed right after the question mark)

File: mobile/src/main/java/org/openhab/habdroid/ui/CloudNotificationAdapter.kt [hunk 2]
```diff
@@ -110,7 +113,11 @@ class CloudNotificationAdapter(context: Context, private val loadMoreListener: (
             val conn = ConnectionFactory.cloudConnection
             if (notification.icon != null && conn != null) {
                 val encodedIcon = Uri.encode(notification.icon)
-                iconView.setImageUrl(conn, "images/$encodedIcon.png",
+                val suffix = when (itemView.context.getPrefs().getIconFormat()) {
+                    IconFormat.Png -> "png"
+                    IconFormat.Svg -> "svg"
+                }
+                iconView.setImageUrl(conn, "icon/$encodedIcon?&format=$suffix&anyFormat=true",
```
Comment (maniac103) [inline mobile/src/main/java/org/openhab/habdroid/ui/CloudNotificationAdapter.kt]: As above.

**Comment by mueller-ma** [comment]
> I wonder whether we should centralize this logic

I was thinking about this as well, but couldn't find a good way to pass both icon name and/or icon path (used by LinkedPage) to the new function.

File: mobile/src/full/java/org/openhab/habdroid/core/FcmMessageListenerService.kt (hunk 1 above)
Comment (mueller-ma) [reply mobile/src/full/java/org/openhab/habdroid/core/FcmMessageListenerService.kt]: Thanks!

**Comment by maniac103** [comment]
@mueller-ma What do you think about https://github.com/maniac103/openhab.android/commit/6e6bc147d397756baeccc6f578f8577fc6390bdb ? I thought about this a bit more and think the problem in finding a good extension function is the inconsistent icon path handling in the app ... hence this is what I came up with. (NB: it's only compile time tested so far)

**Comment by mueller-ma** [comment]
TODO:

- [ ] Fix missing icon if widget is created from within the app
- [ ] Add fallback or migrate existing item update widget data

**Comment by maniac103** [comment]
> Add fallback or migrate existing item update widget data

https://github.com/maniac103/openhab.android/commit/dc6600dcaef63a123052fc41e5e9a6a04900af80

**Comment by mueller-ma** [comment]
Should be RTM now.

**Pull request merged**